### PR TITLE
Setup initial library/binary/benchmarking project layout

### PR DESCRIPTION
**Pull request by colinwahl**
I'm definitely open to suggestions here.

We'd like to use [criterion](https://github.com/bheisler/criterion.rs) for benchmarking as we work on this project - however, criterion can only benchmark `lib` code. The previous project setup had it in a binary.

This PR splits out the crate into two parts - a lib, and a binary. The idea being that the lexer/parser/formatter will live in `lib`, and then we will produce a binary which puts it all together within `bin`.

In this way, we can still get benchmarking/testing for all our lib code as usual while also producing an executable within the same repository.

### CODE REVIEW

File: bin/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "pactfmt"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+glob = "0.3.1"
+pactfmt-lib = {path = "../lib" }
```
Comment (thomashoneyman) [inline bin/Cargo.toml:8]: This spacing looks a bit weird. Was it intentional?

```suggestion
pactfmt-lib = { path = "../lib" }
```

File: lib/benches/benchmark.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use logos::Logos;
+use pactfmt_lib::Token;
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    fn lex(input: &str) -> Vec<Result<Token, ()>> {
+        Token::lexer(input).collect()
+    }
+
+    c.bench_function("lex", |b| b.iter(|| lex(black_box("( )"))));
```
Comment (thomashoneyman) [inline lib/benches/benchmark.rs:10]: This better be quick!

**Comment by thomashoneyman** [comment]
All the workflow jobs are running twice because they're triggering on pull_request and push. I think we should separate these so a job runs on pull_request _or_ push, as that will run on all commits to a pull request and all pushes to main, but not both.

```yaml
on:
  push:
    branches: [main]
  pull_request:
```

File: bin/Cargo.toml (hunk 1 above)
Comment (colinwahl) [reply bin/Cargo.toml]: Ah, good catch.

File: lib/benches/benchmark.rs (hunk 2 above)
Comment (colinwahl) [reply lib/benches/benchmark.rs:10]: I sure hope so :D

**Pull request merged**